2q.go

```diff
@@ -103,11 +103,11 @@ func (c *TwoQueueCache) Get(key interface{}) (value interface{}, ok bool) {
 	}
 
 	// No hit
-	return nil, false
+	return
```

Comment:
Elsewhere in this function it explicitly returns parameters instead of using the named parameters. While this function will work correctly so far as I know, I think it reads more confusing that it needs to. Since you're updating stuff, can you remove the named return parameters (here and in other Get methods) and keep the `return nil, false`? It's more clear/explicit.

Reply:
You could also keep the named params for clarity in documentation but just be explicit in all return clauses.

Reply:
We can't keep `return nil, false`, because `nil` can't be assigned to all possible types of `V` (privimite types, non-pointer structs, etc.). This is also the reason for the other changes to naked returns.

https://go.dev/play/p/tf-1bsx02Q1

I can change the other return statements or add comments if you want.

Reply:
Interesting. Given that `any` is the same as `interface{}`, which itself is a reference to some other type but can always be nil, I wonder why generics behave this way.

For instance it is valid to have `var p interface{}` which is equivalent to `nil`. It is also valid to have `var b *Foobar; var p interface{}; p = b` which would be a nil pointer inside a now not-nil interface.

I guess I assumed that `any` in this context for generics meant the same as `any` in non-generic uses.

So it seems like there are two options: one is accept that some types may return empty instead of nil, document this as a breaking change in v2, and tell people if they want the old semantics to always use correct types; the other is to have generics for K only and just use `any` as the value type in all cases. Probably better to do the former though.

Reply:
I don't think this is a breaking change. Returning empty values is fine because `ok` will be false, indicating that there was no hit and the return value isn't valid. The built-in go map has the exact same behaviour[^1].

We could modify the function signature to make it return a pointer[^2], but I don't really like that approach.

[^1]: https://go.dev/play/p/MohR-gp_zgs
[^2]: https://go.dev/play/p/zLL4NU2tfyX

Reply:
Ah yes, forgot about `ok` there